taxcalc/tests/test_parameters.py

```diff
@@ -28,3 +42,81 @@ def test_ParametersBase_instantiation_and_usage():
     with pytest.raises(ValueError):
         ParametersBase.expand_array(threedarray, True, [0.02, 0.02], 2)
+
+
+@pytest.mark.parametrize("fname",
+                         [("behavior.json"),
+                          ("consumption.json"),
+                          ("current_law_policy.json"),
+                          ("growdiff.json")])
+def test_json_file_contents(tests_path, fname):
+    """
+    Check contents of JSON parameter files.
+    """
+    # pylint: disable=too-many-locals
+    # specify test information
+    reqkeys = ['long_name', 'description', 'notes',
+               'row_var', 'row_label',
+               'start_year', 'cpi_inflated',
+               'col_var', 'col_label',
+               'value']
+    first_year = Policy.JSON_START_YEAR
+    last_known_year = 2017  # for indexed parameters
+    num_known_years = last_known_year - first_year + 1
+    # read JSON parameter file into a dictionary
+    path = os.path.join(tests_path, '..', fname)
+    pfile = open(path, 'r')
+    allparams = json.load(pfile)
+    pfile.close()
+    assert isinstance(allparams, dict)
+    # check elements in each parameter sub-dictionary
+    for pname in allparams:
+        param = allparams[pname]
+        assert isinstance(param, dict)
+        # check that param contains required keys
+        for key in reqkeys:
+            assert key in param
+        # check that row_var is FLPDYR
+        assert param['row_var'] == 'FLPDYR'
+        # check that start_year equals first_year
+        syr = param['start_year']
+        assert isinstance(syr, int) and syr == first_year
+        # check that cpi_inflated is boolean
+        assert isinstance(param['cpi_inflated'], bool)
+        # check that row_label is list starting with first_year
+        rowlabel = param['row_label']
+        assert isinstance(rowlabel, list) and int(rowlabel[0]) == first_year
+        # check type and dimension of value
+        value = param['value']
+        assert isinstance(value, list)
+        assert len(value) == len(rowlabel)
+        # check that col_var and col_label are consistent
+        cvar = param['col_var']
+        assert isinstance(cvar, six.string_types)
+        clab = param['col_label']
+        if cvar == '':
+            assert isinstance(clab, six.string_types) and clab == ''
+        else:
+            assert isinstance(clab, list)
+            # check different possible col_var values
+            if cvar == 'MARS':
+                assert len(clab) == 6
+            elif cvar == 'EIC':
+                assert len(clab) == 4
+            elif cvar == 'idedtype':
+                assert len(clab) == 7
+            else:
+                assert cvar == 'UNKNOWN col_var VALUE'
+            # check length of each value row
+            for valuerow in value:
+                assert len(valuerow) == len(clab)
+        # check that indexed parameters have all known years in rowlabel list
+        form_parameters = ['_AMT_em_pe',
```

Comment:
@martinholmer, I confirmed that 2017 projections for the `form_parameters` are unavailable in James C Young's 
[Inflation Adjustments Affecting Individual Taxpayers in 2017](https://github.com/open-source-economics/Tax-Calculator/files/885889/153tn0251.pdf).